src/promptflow/promptflow/_core/operation_context.py

```diff
@@ -119,11 +125,14 @@ def append_user_agent(self, user_agent: str):
         Args:
             user_agent (str): The user agent information to append.
         """
-        if "user_agent" in self:
-            if user_agent not in self.user_agent:
-                self.user_agent = f"{self.user_agent} {user_agent}"
-        else:
-            self.user_agent = user_agent
+        agent = self.get("_user_agent", '')
+        user_agent = user_agent.strip()
+        if user_agent not in agent:
+            self._user_agent = f"{agent} {user_agent}".strip()
+
+    @property
+    def user_agent(self):
```

Comment:
Why added this?

Reply:
To avoid errors caused by non-existent attributes when using `context.user_agent` directly.
In addition, we must also ensure that the values obtained through `context.user_agent` are consistent with those obtained through `context..get_user_agent().`

![image](https://github.com/microsoft/promptflow/assets/23182548/a4df4d4b-ccd1-4cee-a635-08cf4d22b433)